### PR TITLE
Workaround for Json.Net bug

### PR DESCRIPTION
**Pull request by ajaybhargavb**
@rynowak @Eilon

### CODE REVIEW

**Comment by Eilon** [comment]
Reviewed in person so this looks :shipit: to me, but need @rynowak to confirm that this was the intention.

File: src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/JsonArrayPool.cs [hunk 1]
```diff
@@ -33,7 +39,11 @@ public void Return(T[] array)
                 throw new ArgumentNullException(nameof(array));
             }
 
-            _inner.Return(array);
+            if (_arrayTracker.Contains(array))
+            {
+                _arrayTracker.Remove(array);
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/JsonArrayPool.cs]: Can combine `Contains()` and `Remove()` calls: `if (_arrayTracker.Remove(array))`

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/JsonArrayPool.cs]: :+1:

Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/JsonArrayPool.cs]: Look at me! I used the new reaction thing to agree with @dougbu

**Comment by rynowak** [comment]
:shipit: - as workarounds go, this one is pretty great I gotta say :+1:

**Comment by ajaybhargavb** [comment]
Closing this as it is no longer needed.

**Comment by rynowak** [comment]
We should be able to use a new 8.0.3 package and so this won't be needed.

**Pull request closed without merging**